tests/test_trace_encoder.py

```diff
@@ -59,12 +56,15 @@
 from opentelemetry.trace import TraceFlags as SDKTraceFlags
 from opentelemetry.trace.status import Status as SDKStatus
 from opentelemetry.trace.status import StatusCode as SDKStatusCode
+from snowflake.telemetry._internal.exporter.otlp.proto.traces import (
+    ProtoSpanExporter,
+)
 
 
 class TestOTLPTraceEncoder(unittest.TestCase):
```

Comment:
What do we test here?

Reply:
We're using the mock data helpfully created here to:

1. Make sure that the behavior of our custom 1.12.0-based _encode_spans() method matches the behavior of the new 1.23.0 encode_spans() method from the opentelemetry lib
2. Exercise ProtoSpanExporter._serialize_traces_data() to do a basic sanity check on the bytes it produces

Reply:
Should we then test using our "public" API which is the Exporter?

Reply:
Yes, I've updated the tests to exercise the ProtoSpanExporter, ProtoMetricExporter, and _ProtoLogExporter, respectively.